lib/Sema/CSFix.cpp

```diff
@@ -1148,6 +1148,33 @@ bool AllowArgumentMismatch::diagnose(const Solution &solution,
   return failure.diagnose(asNote);
 }
 
+bool AllowArgumentMismatch::diagnoseForAmbiguity(
+    CommonFixesArray commonFixes) const {
+  // We have a case where the same exact argument mismatch fix is produced
+  // across multiple solutions. So we just diagnose it here.
+  auto front = commonFixes.front();
+  auto firstApplyInfo = front.first->getFunctionArgApplyInfo(front.second->getLocator());
+  if (!firstApplyInfo)
+    return false;
+  
+  if (llvm::all_of(
+          commonFixes,
+          [&](std::pair<const Solution *, const ConstraintFix *> fix) {
+            if (fix.second->getKind() != FixKind::AllowArgumentTypeMismatch)
+              return false;
+            auto *constraintFix = fix.second->getAs<AllowArgumentMismatch>();
+            auto applyInfo =
+                fix.first->getFunctionArgApplyInfo(constraintFix->getLocator());
+            return applyInfo && applyInfo->getFnType()->getNumParams() == 1 &&
```

Comment:
Why is it necessary for the function to only have one parameter?

Reply:
Ah, the intention was to make sure the arguments were the same, maybe its better to just compare the positions :)

Reply:
If we get here it should already be the case that all of the fixes have the same locator

Reply:
Thanks @hborla, I left a more detailed comment below :)